torch/export/exported_program.py

```diff
@@ -1161,6 +1161,28 @@ def _update(
             verifiers=verifiers if verifiers is not None else self.verifiers,
         )
 
+    def _move_to(self, location: Union[torch.device, str, Dict[str, str]]):
+        # move all the tensors and parameters to the new location
+        for k, v in self.state_dict.items():
+            if isinstance(location, dict):
+                location = location[str(v.device)]
+            if isinstance(v, torch.nn.Parameter):
+                self._state_dict[k] = torch.nn.Parameter(v.to(location))
+            else:
+                self._state_dict[k] = v.to(location)
+        # change all the nodes with burnt-in device to the new location
+        for node in self.graph.nodes:
+            if "device" in node.kwargs:
+                kwargs = node.kwargs.copy()
+                kwargs["device"] = location
+                node.kwargs = kwargs
+
+    def move_to(self, location: Union[torch.device, str, Dict[str, str]]):
+        """
+        Moves the ExportedProgram to the specified device.
+        """
+        self._move_to(location)
+
```

Comment:
Maybe I'm dumb, but I find it hard to understand the contract provided by this method.

IMO it should be a pass existing in a free function form. Later if we see a need, we can discuss how to make this a method on EP.